.github/actions/setup-win/action.yml

```diff
@@ -37,13 +37,46 @@ runs:
       run: |
         Add-MpPreference -ExclusionPath $(Get-Location).tostring() -ErrorAction Ignore
 
+    - name: Setup conda
+      shell: bash
+      run: |
+        # Windows conda is baked into the AMI at this location
+        CONDA="C:\Jenkins\Miniconda3\condabin\conda.bat"
+
+        {
+          echo "CONDA_RUN=${CONDA} run --no-capture-output";
+          echo "CONDA_BUILD=${CONDA} run conda-build";
+          echo "CONDA_INSTALL=${CONDA} install";
+        } >> "${GITHUB_ENV}"
+
     - name: Setup Python3
-      uses: actions/setup-python@v4
-      with:
-        python-version: '3.x'
-        check-latest: false
-        cache: pip
-        cache-dependency-path: |
-          **/requirements.txt
-          **/.circleci/docker/requirements-ci.txt
-          **/.github/requirements-gha-cache.txt
+      shell: bash
+      run: |
+        set +e
+
+        PYTHON3=$(${CONDA_RUN} which python3)
+        HAS_PYTHON3=$?
+
+        if [[ "${HAS_PYTHON3}" == "0" ]]; then
```

Comment:
The variable name here feels a bit confusing `HAS_FOO == 0` generally implies FOO is not there, where in this case you mean the opposite.

Perhaps rename that to `PYTHON_MISSING` or invert `HAS_PYTHON` while assigning it

Reply:
Agree, I should have called this `EXIT_CODE` or something to make it clearer.  Fixing it now